cpp/src/converters/renumber.cu

```diff
@@ -34,79 +44,176 @@ gdf_error gdf_renumber_vertices(const gdf_column *src, const gdf_column *dst,
+      if (err != GDF_SUCCESS)
+        return err;
+
+      gdf_column_view(numbering_map, tmp, nullptr, new_size, src->dtype);
+      break;
+    }
+
+  case GDF_INT64:
+    {
+      size_t new_size;
+
+      //
+      //  NOTE: At the moment, we force the renumbered graph to use
+      //        32-bit integer ids.  Since renumbering is going to make
+      //        the vertex range dense, this limits us to 2 billion
+      //        vertices.
+      //
+      //        The renumbering code supports 64-bit integer generation
+      //        so we can run this with int64_t output if desired...
+      //        but none of the algorithms support that.
+      //
+      int64_t *tmp;
+      ALLOC_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
+      gdf_column_view(src_renumbered, tmp, src->valid, src->size, GDF_INT32);
+
+      ALLOC_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
+      gdf_column_view(dst_renumbered, tmp, dst->valid, dst->size, GDF_INT32);
+
+      gdf_error err = cugraph::renumber_vertices(src->size,
+                                                 (const int64_t *) src->data,
+                                                 (const int64_t *) dst->data,
+                                                 (int32_t *) src_renumbered->data,
+                                                 (int32_t *) dst_renumbered->data,
+                                                 &new_size,
+                                                 &tmp,
+                                                 cugraph::HashFunctionObjectInt(hash_size),
+                                                 thrust::less<int64_t>()
+                                                 );
+      if (err != GDF_SUCCESS)
+        return err;
+
+      //
+      //  If there are too many vertices then the renumbering overflows so we'll
+      //  return an error.
+      //
+      if (new_size > 0x7fffffff) {
+        ALLOC_FREE_TRY(src_renumbered, stream);
+        ALLOC_FREE_TRY(dst_renumbered, stream);
+        return GDF_COLUMN_SIZE_TOO_BIG;
+      }
+
+      //
+      //  Note that the numbering_map column matches the type of the input
+      //  columns (we want the numbering map to take us back to the
+      //  original data, so it must match that column type).
+      //
+      gdf_column_view(numbering_map, tmp, nullptr, new_size, src->dtype);
+      break;
+    }
+
+  case GDF_STRING:
+    {
+      size_t new_size;
+
+      int32_t *tmp;
+      ALLOC_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
+      gdf_column_view(src_renumbered, tmp, src->valid, src->size, GDF_INT32);
+
+      ALLOC_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
+      gdf_column_view(dst_renumbered, tmp, dst->valid, dst->size, GDF_INT32);
+
+      NVStrings *srcList = reinterpret_cast<NVStrings*>(src->data);
+      NVStrings *dstList = reinterpret_cast<NVStrings*>(dst->data);
+
+      typedef std::pair<const char *, size_t>   pair_t;
```

Comment:
Are you using pair_t? I can see std::pair<const char *, size_t> right below this code, so if not used, I recommend deleting this (and I generally do not prefer excessive use of typedef or something like "using pair_t = std::pair<const char*, size_t>" to save little bit of typing, I recommend using aliases only when it significantly improves code readability, if I see pair_t, I need additional lookup to figure out pair_t actually means std::pair<const char*, size_t>, but this is just recommendation).

Reply:
Agreed.  One of the intermediate forms of this code made more full use of this, now it is reduced to a typing saver... I will fix this in my next push.